plugins/ssr/ssrclasses.ml

```diff
@@ -0,0 +1,59 @@
+(************************************************************************)
+(*         *   The Coq Proof Assistant / The Coq Development Team       *)
+(*  v      *   INRIA, CNRS and contributors - Copyright 1999-2019       *)
+(* <O___,, *       (see CREDITS file for the list of authors)           *)
+(*   \VV/  **************************************************************)
+(*    //   *    This file is distributed under the terms of the         *)
+(*         *     GNU Lesser General Public License Version 2.1          *)
+(*         *     (see LICENSE file for the text of the license)         *)
+(************************************************************************)
+
+(** Compatibility layer for [under] and [setoid_rewrite]; see [ssrclasses.v]. *)
+
+(* Clone of [Rewrite.find_global] not using [Coqlib.find_reference] (deprec.) *)
+let find_global s =
+  let gr = lazy (Coqlib.lib_ref s) in
+    fun (evd,cstrs) ->
+      let (evd, c) = Evarutil.new_global evd (Lazy.force gr) in
+        (evd, cstrs), c
+
+(* Copy of [Rewrite.app_poly_check] *)
+let app_poly_check env evars f args =
+  let (evars, cstrs), fc = f evars in
+  let evars, t = Typing.solve_evars env evars (EConstr.mkApp (fc, args)) in
+  (evars, cstrs), t
```

Comment:
@gares thanks for your comment; it's true that this code seems to be duplicated, but the precise feature that was needed here was not available in Coq's `Rewrite`: basically I need to use a "clone" of `Rewrite.get_reflexive_proof`, using class `Coq.ssr.ssrclasses.Reflexive` instead of class `Coq.Classes.RelationClasses.Reflexive`, which is hard-coded in `Rewrite` currently:

https://github.com/coq/coq/blob/54689c1c1e1333dd1bf63c619481c2ec99a5762e/plugins/ltac/rewrite.ml#L147-L200

The other solution (generalizing `get_reflexive_proof` to any custom "Reflexivity" class for the sake of this PR #10022) sounded a bit more intrusive to me, because it would require to change the interface of `rewrite.mli`…
but if you prefer that latter solution, a refactoring is still possible − in this PR, or in a later PR: I specifically tried to "isolate" this artifact as much as possible in a separate library `ssrclasses.ml, ssrclasses.mli` to ease one such refactoring if need be.
What do you think?